### DISCUSSION TITLE
How to run from docker

### DISCUSSION THREAD

**Discussion by vanous** [maxjoehnk/Mizer#376] [General]
Hi there!

Interesting project, thank you for all the work. I would like to try it but i do not want to even try to install all required dependencies... :) So i am trying to run the docker image, but... how do i connect to it? Or is this not possible and i must run it from the binary, to get some UI?

Thank you!

**Comment by vanous** [comment]
I managed to run it locally but i get the same result... i can see the port 50000 open, but.. still no idea how to connect to it...

**Reply by maxjoehnk** [reply]
https://github.com/maxjoehnk/Mizer/releases

Here you should be able to see the pre-release 3030 (which is just the build number) and a collection of downloads.
The file just called mizer is a headless build similar to the docker image but mizer.zip is the full fledged application which should contain everything so you can just(tm) run it.

**Comment by maxjoehnk** [comment]
Sadly there is currently no way to connect to the docker hosted version via a remote ui. I've only packaged it as a docker build because I created a project with the desktop version but wanted to run it headless on a server.

There is a packaged build though, sadly Github Actions don't support download of them if you're not a maintainer.
I was meaning to add a basic github (pre-)release action so you would be able to download this build.
I'll try to set this up today. Currently only the Linux x64 build is working though.

**Comment by vanous** [comment]
Thank you very much. Actually, it took me a bit of poking around but i did find a way to get the artifacts even if you didn't "release" them... this might be useful to know if you expect the artifacts not to be available... go to the actions - build workflow, choose a previous run, then click on the number below the "Artifacts" label... see below link and screenshot for last weeks builds:

![image](https://github.com/maxjoehnk/Mizer/assets/3680926/67540ed9-1309-4855-b74c-9dfc563a3d72)

https://github.com/maxjoehnk/Mizer/actions/runs/6517828537#artifacts

![image](https://github.com/maxjoehnk/Mizer/assets/3680926/6a0b87ac-8f61-422d-96bf-ec24e99d2e47)

Back to the builds... would it be possible to also include the built libraries in the zip file? All my system libs are newer (higher versions) then the one that app needs... The docker versions has them all but also at this newer version. I could also try to have the standalone linux to match the docker version and PR it but i presume you keep it this way for your system?

**Reply by maxjoehnk** [reply]
Ah that's nice. I only ever tested this without being logged in and there you're not able to download the artifacts.
But then I can delete the pre-release workflow again.

Which libraries are you talking about? FFmpeg for example is statically linked so i don't have to distribute it. The only problem I had in the past was with some gnome libraries, glib I think, which is why I've built the flatpak. Though the version was to old on the target system in this case.

Usually I'm running the linux-full zip on up-to-date arch which should have newer versions than the ubuntu-22.04 used by github actions though, so this shouldn't be an issue.

Which error do you receive when starting the app?

**Comment by vanous** [comment]
What i am missing are these:

	libavdevice.so.58 => not found
	libavfilter.so.7 => not found
	libavformat.so.58 => not found
	libswscale.so.5 => not found
	libavcodec.so.58 => not found
	libavutil.so.56 => not found

I have offered the newer ones, but mizer doesn't like them :)

./mizer: libavdevice.so.58: version `LIBAVDEVICE_58' not found (required by ./mizer)
./mizer: libavfilter.so.7: version `LIBAVFILTER_7' not found (required by ./mizer)
./mizer: libavformat.so.58: version `LIBAVFORMAT_58' not found (required by ./mizer)
./mizer: libswscale.so.5: version `LIBSWSCALE_5' not found (required by ./mizer)
./mizer: libavcodec.so.58: version `LIBAVCODEC_58' not found (required by ./mizer)
./mizer: libavutil.so.56: version `LIBAVUTIL_56' not found (required by ./mizer)


I will try to look at the CI and add them to the artifacts for me... when time permits.

**Reply by maxjoehnk** [reply]
Ah that's ffmpeg. I'll check the static linking, maybe something broke because the flatpak build wasn't work either earlier today.

**Reply by maxjoehnk** [reply]
Apparently the change to the features of ffmpeg broke the static linking, even though the `static` feature is enabled.

Can you check whether the build of this PR works for you?

https://github.com/maxjoehnk/Mizer/pull/383

**Comment by vanous** [comment]
Thank you! The lib issue is now gone, but i still have no UI...

These seems to be the critical messages, although i am not sure...:

(mizer:4463): Gdk-CRITICAL **: 20:39:33.260: gdk_window_get_state: assertion 'GDK_IS_WINDOW (window)' failed
(mizer:4463): Gtk-CRITICAL **: 20:39:33.383: gtk_window_resize: assertion 'width > 0' failed

See full log attached [log.txt](https://github.com/maxjoehnk/Mizer/files/13209342/log.txt)

Based on the above perhaps this is related: https://github.com/flutter/flutter/issues/134771 , maybe the issue is in older Flutter? Not sure, i don't know the Flutter ecosystem too much. 

I tried running the flatpak (i get missing runtime org.freedesktop.Platform/x86_64/22.08), but i am not sure i would like to try to get the Ubuntu's flatpak on my system.

I tried the linux-full (that gave me the messages as listed above) on X11 and Wayland, on iGPU and dGPU always with the same result.

I also tried the linux-headless, this seems to run in the console, but of course on UI...
Any idea? Thank you!

**Reply by maxjoehnk** [reply]
Are you running a compositor?
There seems to be a bug in flutter (or nativeshell which is the flutter embedder I'm using) on X11 with compositing enabled which causes the ui to render as transparent.
I had this issue on PopOS for example. I'm usually running i3 without a compositor so this issue doesn't cause any issues for me but this could be the issue for you.

Afaik the same issue would also occur in the flatpak build.

If you can, please try whether that's indeed the issue. If it is I'll see what I can do about it.

FYI: The github build is always run with the latest stable flutter version, so any bugs fixed in newer versions should be solved in Mizer as well.

**Reply by vanous** [reply]
Interesting, thank you! Yes, that was it, when running in different WM, the UI is now drawn. Now i can start exploring mizer :) . I can see many projects in the examples folder, this is great!
